manifests/global_hook.pp

```diff
@@ -77,10 +74,6 @@
     fail('gitlab::custom_hook resource must specify either content or source, but not both')
   }
 
-  if $source {
-    validate_re($source, '^puppet:')
-  }
-
   $hook_path = "${_custom_hooks_dir}/${type}.d"
```

Comment:
What does it means with `undef` as default value for `$type` and `$custom_hooks_dir` ?

It looks better to not provide defaults at all and remove `Optional` for parameters `$type` and `$custom_hooks_dir`.

Reply:
`Optional` datatype yet present with `$custom_hooks_dir` ?